src/test/java/io/confluent/connect/elasticsearch/integration/ElasticsearchIntegrationTestBase.java

```diff
@@ -0,0 +1,164 @@
+  public void setUp() throws Exception {
+    client = new JestElasticsearchClient("http://" + container.getHttpHostAddress());
+    converter = new DataConverter(true, BehaviorOnNullValues.IGNORE);
+  }
+
+  @After
+  public void tearDown() {
+    try {
+      client.deleteAll();
+    } catch (IOException ex) {
+      // IGNORE in case of error
+    } catch (java.lang.IllegalStateException illegalStateException) {
+      // IGNORED for now until we can fix
+      // this issue https://stackoverflow.com/questions/25889925
+      // Issue looks to be a race condition with the close method on httpcore 4.4 in shared
+      // environments like test could be. Need more research, but for now it can be ignored
+      // as this is only for the tear down of the test.
+    }
+    if (client != null) {
+      client.close();
+    }
+    client = null;
+  }
+
+
+  protected Struct createRecord(Schema schema) {
+    Struct struct = new Struct(schema);
+    struct.put("user", "Liquan");
+    struct.put("message", "trying out Elastic Search.");
+    return struct;
+  }
+
+  protected Schema createSchema() {
+    return SchemaBuilder.struct().name("record")
+        .field("user", Schema.STRING_SCHEMA)
+        .field("message", Schema.STRING_SCHEMA)
+        .build();
+  }
+
+  protected Schema createOtherSchema() {
+    return SchemaBuilder.struct().name("record")
+        .field("user", Schema.INT32_SCHEMA)
+        .build();
+  }
+
+  protected Struct createOtherRecord(Schema schema) {
+    Struct struct = new Struct(schema);
+    struct.put("user", 10);
+    return struct;
+  }
+
+  protected void verifySearchResults(Collection<SinkRecord> records, boolean ignoreKey, boolean ignoreSchema) throws IOException {
+    verifySearchResults(records, TOPIC, ignoreKey, ignoreSchema);
+  }
+
+  protected void verifySearchResults(Collection<?> records, String index, boolean ignoreKey, boolean ignoreSchema) throws IOException {
+    final JsonObject result = client.search("", index, null);
+
+    final JsonArray rawHits = result.getAsJsonObject("hits").getAsJsonArray("hits");
+
+    assertEquals(records.size(), rawHits.size());
+
+    Map<String, String> hits = new HashMap<>();
+    for (int i = 0; i < rawHits.size(); ++i) {
+      final JsonObject hitData = rawHits.get(i).getAsJsonObject();
+      final String id = hitData.get("_id").getAsString();
+      final String source = hitData.get("_source").getAsJsonObject().toString();
+      hits.put(id, source);
+    }
+
+    for (Object record : records) {
+      if (record instanceof SinkRecord) {
+        IndexableRecord indexableRecord = converter.convertRecord((SinkRecord) record, index, TYPE, ignoreKey, ignoreSchema);
+        assertEquals(indexableRecord.payload, hits.get(indexableRecord.key.id));
+      } else {
+        assertEquals(record, hits.get("key"));
+      }
+    }
+  }
+
+  private static String getElasticsearchContainerVersion() {
+    return System.getProperty("esVersion", DEFAULT_ES_VERSION);
```

Comment:
I just recently created https://github.com/confluentinc/connect-testcontainers for the JMS connectors (there were no existing TestContainer classes for JMS brokers), and there I read from first the system property and then a similarly-named environment property.

I don't know whether it's worth also checking an environment property here, but I would suggest that we adopt a similar naming convention for this project for the system properties. In https://github.com/confluentinc/connect-testcontainers#overriding-the-image-version I outline the approach of using a period-separated all lowercase system property (e.g., `activemq.version` or `es.version` here) because it's easily mapped to underscore-separated uppercase environment variable names (e.g., `ACTIVEMQ_VERSION` or `ES_VERSION` here). 

Thus, my suggestion is to:
```suggestion
    return System.getProperty("es.version", DEFAULT_ES_VERSION);
```